elasticdl/python/master/task_dispatcher.py

```diff
@@ -328,6 +331,17 @@ def report(self, request, success):
 
         return (time.time() - start_time), task, worker_id
 
+    def check_exceed_max_task_retries(self, task):
+        self._task_retry_count.setdefault(task, 1)
```

Comment:
Remove the key 'task' if it's already success. Otherwise the memory will keep increasing if the task number is huge.

Reply:
Done!